### PR TITLE
(Not a fork) Bugfix: read zarr from s3

### LINKED ISSUE #13: Read from S3 with s3fs
### Describe the Bug

Following the s3fs example in [the docs](https://bioio-devs.github.io/bioio/bioio.BioImage.html) I should be able to read a ZARR from S3. Instead, I get an `UnsupportedFileFormatError`

### Reproduction
Minimal example
```python
from bioio import BioImage
import s3fs

s3 = s3fs.S3FileSystem(anon=True)
path = "s3://allencell/aics/nuc_morph_data/data_for_analysis/baseline_colonies/20200323_09_small/raw.ome.zarr"
image = BioImage(path)
```

Additional example with some clues.
```python
# %%
from bioio import BioImage
import bioio_base as biob
from ome_zarr.io import parse_url
from ome_zarr.reader import Reader as ZarrReader
import s3fs

s3 = s3fs.S3FileSystem(anon=True)

path = "s3://allencell/aics/nuc_morph_data/data_for_analysis/baseline_colonies/20200323_09_small/raw.ome.zarr"

# %% Alternate version
image = BioImage(path, fs_kwargs=dict(anon=True))  # raises UnsupportedFileFormatError

# %% Related issue
BioImage.determine_plugin(path)  # raises UnsupportedFileFormatError

# %% More clues
reader = ZarrReader(parse_url(path))  # No error, returns a ZarrReader

(abstract_fs, modified_path) = biob.io.pathlike_to_fs(
    path, enforce_exists=True, fs_kwargs={}
)
path2 = "allencell/aics/nuc_morph_data/data_for_analysis/baseline_colonies/20200323_09_small/raw.ome.zarr"
assert modified_path == path2

ZarrReader(parse_url(path2, mode="r"))  # raises AttributeError: 'NoneType' object has no attribute 'exists'
```

### Environment

- OS Version: Ubuntu 18.04
- bioio-ome-zarr 1.0.0
- bioio 1.0.1
- bioio-base 1.0.0
- s3fs 2023.6.0

### PR DESCRIPTION
**Pull request by pgarrison**
Same as #15 but not from a fork

This pull request resolves #13

# Description of Changes

* `ome-zarr` expects a fully-qualified URI and does not accept AbstractFileSystem object. So we use `fs.unstrip_protocol` to recover the fully-qualified URI that `BioImage` has stripped the protocol from.

### Additional changes
* Improved docstring for Reader
* Added detail to `UnsupportedFileFormatError` error message

# Testing
1. Added a test against a public S3 bucket to this repo (it passes on my machine).
2. Created a new project with the following dependencies and ran the following integration test (this used to error). The integration test requires AWS credentials to be configured (e.g., in the `~/.aws` directory).
```python
from bioio import BioImage
path = "s3://allencell/aics/nuc_morph_data/data_for_analysis/baseline_colonies/20200323_09_small/raw.ome.zarr"
image = BioImage(path)
print(image.get_image_dask_data())
```
```
# Dependencies
"bioio-ome-zarr @ git+https://github.com/bioio-devs/bioio-ome-zarr.git@bugfix/read-zarr-from-s3",
"bioio @ git+https://github.com/bioio-devs/bioio.git@main",
"bioio-base @ git+https://github.com/bioio-devs/bioio-base.git@main"
```

### CODE REVIEW

**Comment by pgarrison** [comment]
The new test is failing but I have not figured out why. I can't reproduce the issue locally

**Comment by toloudis** [comment]
> The new test is failing but I have not figured out why. I can't reproduce the issue locally

Did you see this warning in the build logs?
```
UserWarning: Your installed version of s3fs is very old and known to cause
  severe performance issues, see also https://github.com/dask/dask/issues/10276
```
Maybe compare your local version of s3fs with the one used in the automated build?

**Comment by pgarrison** [comment]
Figured it out -- the underlying ome-zarr packages doesn't support passing `{ anon: True }` to fsspec

**Comment by pgarrison** [comment]
`https://` urls work with this plugin, but do not integrate correctly with `bioio-base`

File: README.md [hunk 1]
```diff
@@ -41,6 +41,15 @@ img = BioImage("my_file.zarr", reader=bioio_ome_zarr.Reader)
 img.data
 ```
 
+### Reading from AWS S3
+To read from private S3 buckets or public buckets using `s3://` paths, [credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html) must be configured. Public buckets can be accessed without credentials by using the `https://` path.
```
[the remark this answers is not in the document]
Comment (pgarrison) [reply README.md:45]: It should but there's an upstream [issue](https://github.com/ome/ome-zarr-py/issues/369)

**Review by toloudis — APPROVED** [review]

File: bioio_ome_zarr/reader.py [hunk 2]
```diff
@@ -65,17 +58,19 @@ def __init__(
...
-        self._zarr = ZarrReader(parse_url(self._path, mode="r")).zarr
+        self._zarr = get_zarr_reader(self._fs, self._path).zarr
         self._physical_pixel_sizes: Optional[types.PhysicalPixelSizes] = None
         self._channel_names: Optional[List[str]] = None
 
     @staticmethod
     def _is_supported_image(fs: AbstractFileSystem, path: str, **kwargs: Any) -> bool:
         try:
-            ZarrReader(parse_url(path, mode="r"))
+            get_zarr_reader(fs, path)
```
[the remark this answers is not in the document]
Comment (pgarrison) [reply bioio_ome_zarr/reader.py:73]: Done in [1e016d7](https://github.com/bioio-devs/bioio-ome-zarr/pull/17/commits/1e016d744c7d74dfb3dd6b7d99f28bef9cb326ed)

**Review by toloudis — APPROVED** [review]

**Review by SeanLeRoy — APPROVED** [review]

**Review by BrianWhitneyAI — APPROVED** [review]

**Pull request merged**